js/views/modals/connectionManagement/Configurations.js

```diff
@@ -103,16 +127,61 @@ export default class extends baseVw {
     }
 
     if (e.reason === 'authentication-failed') {
-      msg = app.polyglot.t('connectionManagement.statusBar.errorAuthFailed', {
-        serverName: e.server.get('name'),
-        errorPreface: '<span class="txB">' +
-          `${app.polyglot.t('connectionManagement.statusBar.errorPreface')}</span>`,
-        links,
-      });
+      if (e.server.get('default')) {
+        // If the default server fails with an auth issue, it's because the cookie token is
+        // invalid. This should never happen and must be a dev error.
+        msg = app.polyglot.t('connectionManagement.statusBar.errorAuthFailedBuiltInServer', {
+          serverName: e.server.get('name'),
+          errorPreface: '<span class="txB">' +
+            `${app.polyglot.t('connectionManagement.statusBar.errorPreface')}</span>`,
+          links,
+        });
+      } else {
+        msg = app.polyglot.t('connectionManagement.statusBar.errorAuthFailed', {
+          serverName: e.server.get('name'),
+          errorPreface: '<span class="txB">' +
+            `${app.polyglot.t('connectionManagement.statusBar.errorPreface')}</span>`,
+          links,
+        });
+      }
     } else if (e.reason === 'canceled') {
       this.$statusBarOuterWrap.addClass('hide');
       this.configViews.forEach(configVw => configVw.setState({ status: 'not-connected' }));
       return;
+    } else if (e.reason === 'tor-not-configured') {
+      msg = app.polyglot.t('connectionManagement.statusBar.errorTorNotConfigured', {
+        serverName: e.server.get('name'),
+        editLink: '<a class="js-editDefaultConfig">' +
+          `${app.polyglot.t('connectionManagement.statusBar.editLink')}</a>`,
+        links,
+      });
+
+      if (!this.$el.is(':visible')) {
+        // If the connection modal is not open, we'll open up the configuration form. The modal
+        // will be opened shortly upon the connection failure. If the user already had the modal
+        // open, we won't auto send them to the config form, since it may interrupt something else
+        // they may be doing.
+        this.trigger('editConfig', {
+          model: this.collection.defaultConfig,
+        });
+      }
+    } else if (e.reason === 'tor-not-available') {
+      msg = app.polyglot.t('connectionManagement.statusBar.errorTorNotAvailable', {
+        serverName: e.server.get('name'),
+        editLink: '<a class="js-editDefaultConfig">' +
+          `${app.polyglot.t('connectionManagement.statusBar.editLink')}</a>`,
+        links,
+      });
+
+      if (!this.$el.is(':visible')) {
+        // If the connection modal is not open, we'll open up the configuration form. The modal
+        // will be opened shortly upon the connection failure. If the user already had the modal
+        // open, we won't auto send them to the config form, since it may interrupt something else
+        // they may be doing.
+        this.trigger('editConfig', {
+          model: this.collection.defaultConfig,
```

Comment:
No, we would only get a 'tor-not-available' error on the default (i.e. built-in) server, because that's the only one that we control and can start-up via the status command to determine if Tor is available. For stand-alon servers, it is the responsibility of the user to start the server in Tor mode. If they forgot to start the Tor browser and they start the server with --tor, then it will fail to start.